### ISSUE TITLE
Component does not rerender after vite build

### ISSUE THREAD

**Issue by taorepoara** [preactjs/signals#508]
I made an SPA with Preact (not React) and it's based on signals.

Local development works fine, but when building the projet with ` vite build` the components do not re-render after signal change.

I tried to add log in `effet` function to check if the signal changes and it works fine after build, but not the render.

Here is my vite config file:
```js
import { defineConfig } from 'vite';
import preact from '@preact/preset-vite';

// https://vitejs.dev/config/
export default defineConfig({
	plugins: [preact()],
});
```

And here is a minimalist example: 

```tsx
import { effect, signal } from "@preact/signals";
import { Component } from "preact";

export const isAdmin = signal(false);

effect(() => {
	console.log("isAdmin changed", isAdmin.value);
});

setTimeout(() => {
	isAdmin.value = true;
}, 1000);

export class Admin extends Component {
	render() {
		console.log("Admin render", isAdmin.value);
		if (isAdmin.value) return "Admin";
		return "Not admin";
	}
}
```

**Comment by XantreGodlike** [comment]
Probably reactivity is not working with class components, you should use function components if you want it to rerender on signal changes

**Comment by taorepoara** [comment]
It does not work better with this version: 

```tsx
import { effect, signal } from "@preact/signals";

export const isAdmin = signal(false);

effect(() => {
	console.log("isAdmin changed", isAdmin.value);
});

setTimeout(() => {
	isAdmin.value = true;
}, 1000);

export function Admin() {
	console.log("Admin render", isAdmin.value);
	if (isAdmin.value) return "Admin";
	return "Not admin";
}
```

**Comment by XantreGodlike** [comment]
Strange. What if you wrap return results with fragments? `<>Admin</>`

**Comment by taorepoara** [comment]
preact: 10.19.3
@preact/signals: 1.2.2
@preact/preset-vite: 2.8.1

I'm running the `vite build` with bun. I'll try with npm.

**Comment by taorepoara** [comment]
> I'm running the `vite build` with bun. I'll try with npm.

Not better

**Comment by taorepoara** [comment]
> Not better

My bad ! I made a mistake in my test. It works fine with npm. I'll check if there is version diffs, because I don't understand why it would affect the build.

**Comment by taorepoara** [comment]
It seems to be fixed in preact 10.19.4

Thanks for your help !
